app/Console/Commands/WaitingLists/CheckMembersMeetActivityRules.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace App\Console\Commands\WaitingLists;
+
+use App\Console\Commands\Command;
+use App\Models\Mship\Account;
+use App\Models\NetworkData\Atc;
+use App\Models\Training\WaitingList;
+use App\Notifications\Training\WaitingListRemovalAdded;
+use App\Notifications\Training\WaitingListRemovalCancelled;
+use App\Notifications\Training\WaitingListRemovalCompleted;
+use App\Notifications\Training\WaitingListRemovalReminder;
+use Carbon\Carbon;
+
+class CheckMembersMeetActivityRules extends Command
+{
+
+    protected $minutesRequired = 720; // 12 hours is represented as 720 minutes
+
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'waitinglists:hourchecker';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Checks ATC hour totals for members who are on an ATC waiting list and marks inactive members for automatic removal from the list.';
+
+    /**
+     * A local cache of account activity to prevent duplicate calculations of a members hours
+     * 
+     * @var array
+     */
+    protected $accountActivityCache = [];
+
+    /**
+     * Executes all necessary console commands.
+     */
+    public function handle()
+    {
+        foreach($this->retrieveHourEnforcedWaitingLists() as $waitingList){
+            $this->identifyNewNonEligibleWaitingListAccounts($waitingList);
+            $this->processWaitingListAccountsMarkedForRemoval($waitingList);
+       }
+       
+    }
+
+    /**
+     * Retrieves training waiting lists which enforce hour requirements
+     */
+    protected function retrieveHourEnforcedWaitingLists()
+    {
+        return WaitingList::where('enforce_hour_requirement', '=', 'yes')->get();
+    }
+
+    /**
+     * Identifies accounts on enforced waiting lists who have fallen below the hour requirement
+     * 
+     * @param WaitingList $waitingList
+     */
+    protected function identifyNewNonEligibleWaitingListAccounts(WaitingList $waitingList)
+    {
+        // We only want to find Active waiting list accounts which are not already marked for removal
+        $waitingListAccounts                                        = $waitingList->accounts
+            ->filter(function($account){
+                return (
+                    $account->pivot->current_status->name == 'Active' &&
+                    (
+                        is_null($account->pivot->pending_removal->removal_date ?? null) ||
```

Comment:
```suggestion
                        !!$account->pivot->pending_removal->removal_date ||
```
A null value evaluates to falsy, so the additional logic is unnecessary.

Reply:
If there is not a pending removal row for this waiting list account, then you will be trying to access removal_date on null. The same could be achieved by doing $account->pivot->pending_removal?->removal_date resulting in slightly less code than the current implementation.